Makefile BINDIR is hard-coded to /usr/local/bin

<!-- Please only use this template for submitting feature requests -->

#### What would you like to be added:
The possibility to install cri-tools in other directories and not only in `/usr/local/bin`, which is hard-coded as `BINDIR := /usr/local/bin` in `Makefile`.

#### Why is this needed:
Allowing cri-tools to be installed on other directories, e.g.: `/opt/bin`, adds flexibility to the installation process without changing previous behavior.  The default installation directory can still be `/usr/local/bin` if `BINDIR` is not defined by user.

Hey @saschagrunert :)

It does work:

```
$ make install BINDIR=/tmp
install -D -m 755 /home/muriloo/sources/cri-o/cri-tools/_output/critest /tmp/critest
install -D -m 755 /home/muriloo/sources/cri-o/cri-tools/_output/crictl /tmp/crictl
```

However, it doesn't honor the `BINDIR` environment variable:

```
$ export BINDIR=/tmp
$ make install
install -D -m 755 /home/muriloo/sources/cri-o/cri-tools/_output/critest /usr/local/bin/critest
install: cannot create regular file '/usr/local/bin/critest': Permission denied
make: *** [Makefile:79: install-critest] Error 1
```

Using a deferred assignment for `BINDIR`, makes it more flexible and also honor the environment:

```
$ export BINDIR=/tmp
$ make install
install -D -m 755 /home/muriloo/sources/cri-o/cri-tools/_output/critest /tmp/critest
install -D -m 755 /home/muriloo/sources/cri-o/cri-tools/_output/crictl /tmp/crictl
```

--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@ ifeq ($(GOOS),windows)
 endif
 
 PROJECT := github.com/kubernetes-sigs/cri-tools
-BINDIR := /usr/local/bin
+BINDIR ?= /usr/local/bin
 
 VERSION := $(shell git describe --tags --dirty --always)
 VERSION := $(VERSION:v%=%)

